### PR TITLE
sbom generation using sbom-tool

### PR DESCRIPTION
**Pull request by viveksahu26**
closes https://github.com/interlynk-io/sbomasm/issues/93

The SBOM is currently generated from the binary using the sbom section of GoReleaser. [goreleaser](https://goreleaser.com/blog/supply-chain-security/) utilizes the syft tool for SBOM creation, but due to issues with syft, we are transitioning to [sbom-tool](https://github.com/microsoft/sbom-tool) for SBOM generation in our CI/CD pipeline.

### CODE REVIEW

**Comment by viveksahu26** [comment]
Check this release: https://github.com/viveksahu26/sbomasm/releases/tag/v1.0.9

**Review by riteshnoronha — CHANGES_REQUESTED** [review]
@viveksahu26 i think these changes are incorrect. The change should only be in https://github.com/interlynk-io/sbomasm/blob/main/.github/workflows/release.yml. 

After the make release section is run, we need to produce an sbom using sbom-tool, and upload it to the assessts section of https://github.com/interlynk-io/sbomasm/blob/main/.github/workflows/release.yml

Please fork the repo test the process and let me know.

**Comment by viveksahu26** [comment]
See my  v1.0.10 release on forked repo: https://github.com/viveksahu26/sbomasm/releases

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -22,17 +35,59 @@ jobs:
...
-      - name: Download syft binary
-        run: curl -sSfL https://raw.githubusercontent.com/anchore/syft/main/install.sh | sh -s -- -b /usr/local/bin
-      - name: Run syft
-        run: syft version
+      
+      - name: Get Tag
+        id: get_tag
+        run: echo "LATEST_TAG=$(git describe --tags --abbrev=0 2>/dev/null || echo 'v0.0.1')" >> $GITHUB_ENV
+
+      - name: Set up Python
```
Comment (riteshnoronha) [inline .github/workflows/release.yml:43]: Why do we need this, we are not pushing this SBOM to interlynk. Thats already done in other actions.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -22,17 +35,59 @@ jobs:
...
+          install-only: true
+
       - run: go version
       - run: goreleaser -v 
       - name: Releaser
         run: make release 
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout Python SBOM tool
```
Comment (riteshnoronha) [inline .github/workflows/release.yml:60]: We dont need this.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -22,17 +35,59 @@ jobs:
...
+          SANITIZED_REF=$(echo "${{ github.ref_name}}" | sed -e 's/[^a-zA-Z0-9.-]/-/g' -e 's/^[^a-zA-Z0-9]*//g')
+          VERSION= ${{ env.LATEST_TAG }}-$SANITIZED_REF          
+          $RUNNER_TEMP/sbom-tool generate -b ${{ env.SBOM_TEMP_DIR }} -bc . -pn ${{ env.TOOL_NAME }} -pv $VERSION -ps ${{ env.SUPPLIER_NAME}} -nsb ${{ env.SUPPLIER_URL }} -cd "--DirectoryExclusionList ${{ env.MS_SBOM_TOOL_EXCLUDE_DIRS }}"
+      
+      - name: Upload SBOM as Release Asset
+        uses: actions/upload-release-asset@v1
+        with:
+          upload_url: ${{ github.event.release.upload_url }}
+          asset_path: ${{ env.SBOM_FILE_PATH }}
+          asset_name: sbomasm.sbom.spdx.json
```
Comment (riteshnoronha) [inline .github/workflows/release.yml:90]: this name should be sbomasm-v015.spdx.json so <tool-name>-<version>.spdx.json

**Review by riteshnoronha — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (viveksahu26) [reply .github/workflows/release.yml:43]: Okay.

**Review by riteshnoronha — APPROVED** [review]

**Pull request merged**